### ISSUE TITLE
S24_OneUI7.0 update bluetooth patch

### ISSUE THREAD

**Issue by jrkruse** [3arthur6/BluetoothLibraryPatcher#74]
needs updated for latest hex patch needs updated
[BluetoothLibPatcher-files_2.zip](https://github.com/user-attachments/files/18046017/BluetoothLibPatcher-files_2.zip)

**Comment by 3arthur6** [comment]
[BluetoothLibraryPatcher_module_2.7.2.zip](https://github.com/user-attachments/files/18047954/BluetoothLibraryPatcher_module_2.7.2.zip)

Is this one working for you?

**Comment by J0SH1X** [comment]
This one flashes but doesn't work after reboot my bt devices are still gone on s23 one ui 7

**Comment by 3arthur6** [comment]
It sounds bad as the files provided show some big compared to the last four one ui updates...

**Comment by 3arthur6** [comment]
> This one flashes but doesn't work after reboot my bt devices are still gone on s23 one ui 7

Could you upload here your /data/misc/bluedroiddump folder?

**Comment by J0SH1X** [comment]
Sure give me 10 minutes

**Comment by J0SH1X** [comment]
> /data/misc/bluedroiddump

the directory is empty

**Comment by 3arthur6** [comment]
> > /data/misc/bluedroiddump
> 
> the directory is empty

Damn.. it will be more hard so. Could you upload the magisk module install log?

**Comment by J0SH1X** [comment]
where would that be located

**Comment by 3arthur6** [comment]
> where would that be located

Install the module again and at the end you have an icon in the top right to save the installation log.

> [subBuffer.log](https://github.com/user-attachments/files/18048333/subBuffer.log) i got the log

It looks like the module is not installed or doesn't have any effect..

**Comment by J0SH1X** [comment]
> > where would that be located
> 
> Install the module again and at the end you have an icon in the top right to save the installation log.
> 
> > [subBuffer.log](https://github.com/user-attachments/files/18048333/subBuffer.log) i got the log
> 
> It looks like the module is not installed or doesn't have any effect..

[magisk_install_log_2024-12-07T13.51.18.log](https://github.com/user-attachments/files/18048383/magisk_install_log_2024-12-07T13.51.18.log)

**Comment by 3arthur6** [comment]
> 6765744B657949564C656E203D207B7D 6765744B657949564C656E203D203136
> 
> From: "getKeyIVLen = {}" To: "getKeyIVLen = 16"
> 
> This should fix the issue

Wow sure? Did you try it?

**Comment by 3arthur6** [comment]
[BluetoothLibraryPatcher_module_2.7.2.zip](https://github.com/user-attachments/files/18049112/BluetoothLibraryPatcher_module_2.7.2.zip)

This one follows @bklynali patch

**Comment by J0SH1X** [comment]
Testing

**Comment by J0SH1X** [comment]
Meh can't upload logs from phone.
Doesn't work still

**Comment by jrkruse** [comment]
> [BluetoothLibraryPatcher_module_2.7.2.zip](https://github.com/user-attachments/files/18049112/BluetoothLibraryPatcher_module_2.7.2.zip)
> 
> This one follows @bklynali patch

this worked for me

**Comment by 3arthur6** [comment]
@J0SH1X you sure it doesnt work? As I get Install the module be sure to reboot then try pairing and reboot.

> > Meh can't upload logs
> > from phone. Doesn't work still
> 
> Try this
> 
> [BluetoothLibraryPatcher_module_2.7.2A.zip](https://github.com/user-attachments/files/18049934/BluetoothLibraryPatcher_module_2.7.2A.zip)
> 
> based on the log I added few more

Hey only hex string 135 is applied with the patch. It relies to the android api, 35 with one ui 7.

**Comment by J0SH1X** [comment]
Yeah, i think it may be a port rom iusse cause it still doesnt work despite seeing reports of it working on s24

**Comment by jrkruse** [comment]
The Bluetooth issue is very odd sometimes it works  and remember pairings with no patch needed. Other times you install and it doesn’t remember pairings and needs the patch module. You got me

**Comment by J0SH1X** [comment]
[subBuffer.log](https://github.com/user-attachments/files/18053997/subBuffer.log)
@bklynali

**Comment by jrkruse** [comment]
I didn’t mean to close issue sorry

**Comment by 3arthur6** [comment]
Wow it never needed as much patches..
What did you changes actually?

**Comment by 3arthur6** [comment]
Wow it never needed as much patches.
What did you changes actually?

**Comment by Mark8980** [comment]
same issue on S25ultra. BT connection working with buds 3 pro though it isnt working with galaxy watch 5. connection lost after rebooting the phone

**Comment by 3arthur6** [comment]
I will take a look on the official OneUI7 firmware

**Comment by 3arthur6** [comment]
@Mark8980 
Is this one fixing the issue?

[BluetoothLibraryPatcher_module_2.7.3.zip](https://github.com/user-attachments/files/18876136/BluetoothLibraryPatcher_module_2.7.3.zip)

**Comment by Mark8980** [comment]
yes this one is working for me. thanks a lot
